### ISSUE TITLE
Pact tags containing slashes cause PactBrokerClient to not fetch them properly

### ISSUE THREAD

**Issue by andreschaffer** [DiUS/pact-jvm#1027]
Hi,

I noticed when one publishes a pact to the pact broker tagged with a value that contains slashes it later on causes an error in the PactBrokerClient when trying to fetch it.

E.g. Consumer: my-consumer published a contract with Provider: my-provider and Tag: feature/yada-yada .

When running my-provider verification tests using the @PactBroker annotation and tag feature/yada-yada it fails with the message:  "au.com.dius.pact.provider.junit.PactRunner: Did not find any pact files for provider my-provider" .

Debugging it, this line https://github.com/DiUS/pact-jvm/blob/master/core/pact-broker/src/main/kotlin/au/com/dius/pact/core/pactbroker/PactBrokerClient.kt#L106 throws an exception "No HAL document found at path 'http://localhost/pacts/provider/my-provider/latest/feature%2Fyada-yada'" .

Checking the path on a browser one can see a response, so something in how the client tries to interpret the response (as HAL)?

I am using au.com.dius:pact-jvm-provider-junit:4.0.6 and dius/pact-broker:2.49.0-1.

Thanks for your help!

**Comment by uglyog** [comment]
I am not able to replicate this. I've tried with both the Gradle plugin and using a JUnit test, and it all worked ok.

If you run the test with DEBUG level logging enabled, you should be able to see the HTTP requests being made. For instance, with my test I can see

```
12:06:09.324 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "GET /pacts/provider/Activity%20Service/latest/tag%2Fwith%2Fslash HTTP/1.1[\r][\n]"
12:06:09.324 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Content-Type: application/json[\r][\n]"
12:06:09.324 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Accept: application/hal+json, application/json[\r][\n]"
12:06:09.324 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Host: test.pact.dius.com.au[\r][\n]"
12:06:09.324 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Connection: Keep-Alive[\r][\n]"
12:06:09.324 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "User-Agent: Apache-HttpClient/4.5.5 (Java/11.0.6)[\r][\n]"
12:06:09.324 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Accept-Encoding: gzip,deflate[\r][\n]"
12:06:09.324 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "Authorization: Basic ZFhmbHR5Rk1nTk9GWkF4cjhpbzl3SjM3aVVwWTQyTTpPNUFJWld4ZWxXYkx2cU1kOFBrQVZ5Y0JKaDJQc3lnMQ==[\r][\n]"
12:06:09.324 [main] DEBUG org.apache.http.wire - http-outgoing-0 >> "[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "HTTP/1.1 200 OK[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Date: Sat, 07 Mar 2020 01:06:09 GMT[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Content-Type: application/hal+json;charset=utf-8[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Content-Length: 1475[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Connection: keep-alive[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Status: 200 OK[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Vary: Accept[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Pact-Broker-Git-Sha: 2da3657410637ead22c0a36b1c75c9ac952278f2[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Request-Id: 178321a7dbcf8840bdbe4a931aaf7f05[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Saas-Broker-Git-Sha: f9f2ccf814bacaa7030f40503bf3ab426c63754e[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Pact-Broker-Version: 2.50.1[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Frame-Options: SAMEORIGIN[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-XSS-Protection: 1; mode=block[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "X-Content-Type-Options: nosniff[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Referrer-Policy: strict-origin[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "Strict-Transport-Security: max-age=31536000 ; includeSubDomains[\r][\n]"
12:06:09.424 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "[\r][\n]"
12:06:09.425 [main] DEBUG org.apache.http.wire - http-outgoing-0 << "{"_links":{"self":{"href":"https://test.pact.dius.com.au/pacts/provider/Activity%20Service/latest/tag%2Fwith%2Fslash","title":"Latest pact versions for the provider Activity Service with consumer version tag 'tag/with/slash'"},"pb:provider":{"href":"https://test.pact.dius.com.au/pacticipants/Activity%20Service","name":"Activity Service"},"pb:pacts":[{"href":"https://test.pact.dius.com.au/pacts/provider/Activity%20Service/consumer/Foo%20Web%20Client/version/1.0.2","title":"Pact between Foo Web Client (1.0.2) and Activity Service","name":"Foo Web Client"},{"href":"https://test.pact.dius.com.au/pacts/provider/Activity%20Service/consumer/Foo%20Web%20Client%202/version/1.0.2","title":"Pact between Foo Web Client 2 (1.0.2) and Activity Service","name":"Foo Web Client 2"}],"provider":{"href":"https://test.pact.dius.com.au/pacticipants/Activity%20Service","title":"Activity Service","name":"DEPRECATED - please use the pb:provider relation"},"pacts":[{"href":"https://test.pact.dius.com.au/pacts/provider/Activity%20Service/consumer/Foo%20Web%20Client/version/1.0.2","title":"DEPRECATED - please use the pb:pacts relation. Pact between Foo Web Client (1.0.2) and Activity Service","name":"Foo Web Client"},{"href":"https://test.pact.dius.com.au/pacts/provider/Activity%20Service/consumer/Foo%20Web%20Client%202/version/1.0.2","title":"DEPRECATED - please use the pb:pacts relation. Pact between Foo Web Client 2 (1.0.2) and Activity Service","name":"Foo Web Client 2"}]}}"
```

that is is working with `tag/with/slash`

**Comment by andreschaffer** [comment]
Hey Ronald,

Thanks for looking into it.

I spent some more time on it after your reply and managed to make it work in my setup.
Indeed the pact library is working correctly.

For the reference in case anyone sees themselves facing a similar problem: what fixed the problem for me in the end was updating dropwizard from 1.3.14 to 2.0.2.
I didn't spend the time to pinpoint exactly the diff and reason why that scenario was not working with dropwizard 1.3.14 for me, but it seems something related to the combination of dependencies / how they were resolved in my project and them not playing well together.

I am closing the issue. Thanks again and keep up the good work!
